### PR TITLE
Apply intelij inspect suggestions for Java code

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/export/README.md [hunk 1]
```diff
@@ -60,10 +60,10 @@ public final class PushExporter {
 **Pull backend:**
 
 ```java
-import io.opentelemetry.sdk.metrics.export.MetricExporter;
+import io.opentelemetry.sdk.metrics.data.MetricData;import io.opentelemetry.sdk.metrics.export.MetricExporter;
```
Comment (jkwatson) [inline sdk/src/main/java/io/opentelemetry/sdk/metrics/export/README.md:63]: something weird here. 2 imports on one line

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/export/README.md [hunk 2]
```diff
@@ -60,10 +60,10 @@ public final class PushExporter {
...
 import io.opentelemetry.sdk.metrics.export.MetricProducer;
 import java.util.ArrayList;
-import java.util.Collections;
+import java.util.Collection;import java.util.Collections;
```
Comment (jkwatson) [inline sdk/src/main/java/io/opentelemetry/sdk/metrics/export/README.md:66]: same here

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/export/README.md [hunk 3]
```diff
@@ -101,6 +101,10 @@ public final class PullExporter {
 
   private void OnPullRequest() {
     // Iterate over all producers and the PullMetricExporter and export all metrics.
+    for (MetricProducer metricProducer : producers) {
```
Comment (carlosalberto) [inline sdk/src/main/java/io/opentelemetry/sdk/metrics/export/README.md:104]: Unrelated change?

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/export/README.md [hunk 4]
```diff
@@ -101,6 +101,10 @@ public final class PullExporter {
 
   private void OnPullRequest() {
```
Comment (jkwatson) [inline sdk/src/main/java/io/opentelemetry/sdk/metrics/export/README.md:102]: just noticed this capitalized method name, also.

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/export/README.md (hunk 3 above)
Comment (bogdandrutu) [reply sdk/src/main/java/io/opentelemetry/sdk/metrics/export/README.md:106]: I was looking to use the producers :)

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/export/README.md (hunk 1 above)
Comment (bogdandrutu) [reply sdk/src/main/java/io/opentelemetry/sdk/metrics/export/README.md]: Done.

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/export/README.md (hunk 2 above)
Comment (bogdandrutu) [reply sdk/src/main/java/io/opentelemetry/sdk/metrics/export/README.md]: Done.

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/export/README.md (hunk 4 above)
Comment (bogdandrutu) [reply sdk/src/main/java/io/opentelemetry/sdk/metrics/export/README.md]: Done.
